legacy/core/src/main/java/com/fsck/k9/preferences/DefaultGeneralSettingsManager.kt

```diff
@@ -152,6 +157,20 @@ internal class RealGeneralSettingsManager(
 
         return settings
     }
+
+    override fun addListener(listener: GeneralSettingsChangeListener) {
+        listeners.add(listener)
+    }
+
+    override fun removeListener(listener: GeneralSettingsChangeListener) {
+        listeners.remove(listener)
+    }
+
+    private fun notifyListeners() {
+        for (listener in listeners) {
+            listener.onGeneralSettingsChanged()
+        }
+    }
```

Comment:
I think it would make sense to extract this into it's own class to allow reuse in feature configurations to avoid the need to always depend on `GeneralSettingsManager` when interested in change notifications.